lib/telemetry_metrics_appsignal.ex

```diff
@@ -108,27 +108,33 @@ defmodule TelemetryMetricsAppsignal do
     metrics = Keyword.get(config, :metrics, [])
 
     Enum.each(metrics, fn metric ->
-      if value = prepare_metric_value(metric, measurements) do
+      if value = prepare_metric_value(metric, measurements, metadata) do
         tags = prepare_metric_tags(metric, metadata)
         send_metric(metric, value, tags)
       end
     end)
   end
 
-  defp prepare_metric_value(metric, measurements)
+  defp prepare_metric_value(metric, measurements, metadata)
 
-  defp prepare_metric_value(%Counter{}, _measurements), do: 1
+  defp prepare_metric_value(%Counter{}, _measurements, _metadata), do: 1
 
-  defp prepare_metric_value(%{measurement: convert}, measurements) when is_function(convert) do
+  defp prepare_metric_value(%{measurement: convert}, measurements, metadata)
+       when is_function(convert, 2) do
+    convert.(measurements, metadata)
+  end
+
+  defp prepare_metric_value(%{measurement: convert}, measurements, _metadata)
+       when is_function(convert) do
```

Comment:
Let's be more specific about the arity here, just in case someone decides to shuffle the definitions around:

```suggestion
       when is_function(convert, 1) do
```

Reply:
I get the intention behind this ask and I would prefer being explicit with the function arities the library supports. However I had a couple thoughts about it: 

First, shuffling the functions around would cause the "binary" function test to fail. That limits the risk of someone accidentally breaking this feature.

In addition, in the present state a `:badarity` error would be raised if a function with an unsupported arity was used. If we added the arity argument then we'd hit the base clause. So the error would be hidden in a sense and that might make problems harder to debug. 

In my view, this is undesirable. Anecdotally, the badarity argument is how I discovered that this library didn't yet support binary function.

What do you think?

Reply:
On the one hand both [`Telemetry.Metrics.Statsd`](https://github.com/beam-telemetry/telemetry_metrics_statsd/blob/main/lib/telemetry_metrics_statsd/event_handler.ex#L114) and [`Telemetry.Metrics.ConsoleReporter`](https://github.com/beam-telemetry/telemetry_metrics/blob/main/lib/telemetry_metrics/console_reporter.ex#L139) match on the exact function arity, but on the other hand the ["Writing Reporters"](https://hexdocs.pm/telemetry_metrics/writing_reporters.html#reacting-to-events) guide suggests either throwing an error and detaching the event handler, or logging an error and keeping it.

In the two libraries mentioned, misconfiguring the reporter will prevent metrics from being sent, but there will be no error and the user will need to debug why aren't metrics being sent. After you explained it, I like your approach better - let's keep it 👍